### ISSUE TITLE
Consider only computing metrics for the validation set.

### ISSUE THREAD

**Issue by dfalbel** [mlverse/luz#83]
We might want to consider only computing the metrics in the validation set by default.
FastAI seems to do it by default, but allows one to specify metrics that should be computed in the training set. 

Callbacks like MixUp and others might modify the inputs and targets in order to implement a technique and this can break metrics in the training set. 

If we don't want this by default, we should at least provide a way to disable metrics from being computed in the validation or training sets.
